### PR TITLE
Issue #3674: Installer: Allow using .po translation files from b.org …

### LINKED ISSUE #3674: [UX] Installer: Allow using .po translation files from b.org without having to rename them.
The translation files available from https://localize.backdropcms.org/translate/downloads are in the format `backdropcms-VERSION.LANGCODE.po`. As in https://github.com/backdrop/backdrop-issues/issues/2345 for files from d.org, we should be accounting for these filenames when scanning for language files during installation, instead of expecting people to rename files to match the pattern `install.LANGCODE.po`.

**Steps To Reproduce**
1. Download a language file from the translation server.
2. Move the downloaded `.po` file, e.g. `backdropcms-1.12.2.es.po`, inside the directory `files/translations` of your Backdrop directory.
3. Install Backdrop as per usual.
4. During the first installation step, you will be prompted to choose a language.

**Actual behavior**
You can't choose another language but English.

**Expected behavior**
You can choose your language, e.g. Spanish.

**Additional information**
The issue had already been [fixed for Drupal files](https://github.com/backdrop/backdrop-issues/issues/2345) before we started to work with our own language server.

---

PR by @olafgrabienski: https://github.com/backdrop/backdrop/pull/2583

### PR DESCRIPTION
**Pull request by olafgrabienski**
…without having to rename them

Fixes https://github.com/backdrop/backdrop-issues/issues/3674.

### CODE REVIEW

File: core/includes/install.core.inc [hunk 1]
```diff
@@ -1304,7 +1304,7 @@ function install_find_translation_files($langcode = NULL) {
   $directory = settings_get('locale_translate_file_directory', conf_path() . '/files/translations');
   // @todo: Remove the "drupal" check in 2.x (assuming we have a localization
   //   server by then).
-  $files = file_scan_directory($directory, '!(install|drupal-\d+\.\d+)\.' . (!empty($langcode) ? preg_quote($langcode, '!') : '[^\.]+') . '\.po$!', array('recurse' => FALSE));
+  $files = file_scan_directory($directory, '!(install|drupal-\d+\.\d+|backdropcms-\d+\.\d+\.\d+)\.' . (!empty($langcode) ? preg_quote($langcode, '!') : '[^\.]+') . '\.po$!', array('recurse' => FALSE));
```
Comment (quicksketch) [inline core/includes/install.core.inc]:
```suggestion
  $files = file_scan_directory($directory, '!(install|drupal-\d+\.\d+|backdrop(cms)?-\d+\.\d+\.\d+)\.' . (!empty($langcode) ? preg_quote($langcode, '!') : '[^\.]+') . '\.po$!', array('recurse' => FALSE));
```

Comment (quicksketch) [reply core/includes/install.core.inc]: To allow the "cms" part of the filename to be optional.

**Review by herbdool — COMMENTED** [review]
A good suggestion. I don't have permission to apply it to @olafgrabienski's repo

File: core/includes/install.core.inc (hunk 1 above)
Comment (olafgrabienski) [reply core/includes/install.core.inc]: Thanks for the suggestion, sounds good, I can make the change Monday or Tuesday.

Comment (olafgrabienski) [reply core/includes/install.core.inc]: Or wait, can I just use the the button "Commit changes" in the "Commit suggestion" dropdown?

Comment (herbdool) [reply core/includes/install.core.inc]: Yes! 😊

**Comment by quicksketch** [comment]
Thanks! Merged into 1.x and 1.12.x.

**Pull request merged**